File: json/clojure/read/json_read.clj
; -----------------------------------------------------------------
;
;	json_read.clj
;
;					Jul/03/2011
;
; -----------------------------------------------------------------
(use '[clojure.contrib.json.read])
(use '[clojure.contrib.duck-streams])
(use '[clojure.contrib.str-utils :only (re-split)])
(use '[clojure.contrib.str-utils :only (str-join)])
; -----------------------------------------------------------------
(load-file "/var/www/data_base/common/clojure_common/text_manipulate.clj")
; -----------------------------------------------------------------
(println "*** 開始 ***")
(def file_json (first *command-line-args*))
;
(def str_json (slurp file_json))
(def dict_aa (read-json str_json))
;
(doseq [unit_aa dict_aa]
	(def key_aa (first unit_aa))
	(def unit_body (second unit_aa))
	(def name_aa (unit_body "name"))
	(def population (unit_body "population"))
	(def date_mod (unit_body "date_mod"))
	(println (str-join "\t" [key_aa name_aa population date_mod]))
)
; -----------------------------------------------------------------
(println (type dict_aa))
;
(println "*** 終了 ***")
; -----------------------------------------------------------------

File: json/ringo/read/json_read.js
#! /usr/bin/ringo
// ------------------------------------------------------------------
//
//	json_read.js
//
//						Dec/27/2013
//
// ------------------------------------------------------------------
var fs= require ("fs");
var text_manipulate = require ('/var/www/data_base/common/ringo_common/text_manipulate.js');
// ------------------------------------------------------------------

print	("*** 開始 ***");

var args = require('system').args;
var file_in = args[1];

var json_str = fs.read (file_in);

var dict_aa = JSON.parse (json_str);

text_manipulate.dict_display_proc (dict_aa);

print	("*** 終了 ***");
// ------------------------------------------------------------------
